### PR TITLE
correct parse_args namespace attribute

### PR DESCRIPTION
**Pull request by PrajwalM2212**
fixes https://github.com/python/typeshed/issues/2366

This PR allows `parse_args` `namespace` arg to accept any object and also to return any object

### CODE REVIEW

File: stdlib/2and3/argparse.pyi [hunk 1]
```diff
@@ -121,8 +121,14 @@ class ArgumentParser(_AttributeHolder, _ActionsContainer):
                      argument_default: Optional[_Text] = ...,
                      conflict_handler: _Text = ...,
                      add_help: bool = ...) -> None: ...
+    @overload
     def parse_args(self, args: Optional[Sequence[_Text]] = ...,
                    namespace: Optional[Namespace] = ...) -> Namespace: ...
+    _N = TypeVar('_N')
+    @overload
+    def parse_args(self, args: Optional[Sequence[_Text]] = ...,
+                   namespace: Type[_N] = ...) -> _N: ...
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply stdlib/2and3/argparse.pyi]: When I remove the default by setting  `namespace: _N` , pycharm  high-lightens it as `non-default parameter follows default parameter`.  Should I ignore it ?

File: stdlib/2and3/argparse.pyi [hunk 2]
```diff
@@ -121,8 +121,16 @@ class ArgumentParser(_AttributeHolder, _ActionsContainer):
                      argument_default: Optional[_Text] = ...,
                      conflict_handler: _Text = ...,
                      add_help: bool = ...) -> None: ...
+
+    @overload
     def parse_args(self, args: Optional[Sequence[_Text]] = ...,
                    namespace: Optional[Namespace] = ...) -> Namespace: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/argparse.pyi]: I think this overload shouldn't have the `namespace` argument. If `namespace` is given, the second overload suffices.

**Comment by srittau** [comment]
@PrajwalM2212 Thank you for your effort to fix those small bugs, I really appreciate it! I think you were nearly there with this PR. Could you try this:

```python
    _N = TypeVar('_N')
    @overload
    def parse_args(self, args: Optional[Sequence[_Text]] = ...) -> Namespace: ...
    @overload
    def parse_args(self, args: Optional[Sequence[_Text]], namespace: _N) -> _N: ...
    @overload
    def parse_args(self, *, namespace: _N) -> _N: ...
```

This changes two things compared to your last version:
* It moved the typevar before the first overload. It seems type checkers don't like it if it's between the overloads. You could also move the typevar to the top of the file, which we usually do
* Remove `Type` around the `namespace` argument (see my previous comment).

**Comment by srittau** [comment]
I am sorry that it failed again. It seems that pytype does not like typevars inside classes. I didn't know that. So just move it to the start of the file and then it should finally pass! Thank you for your persistence.

**Comment by PrajwalM2212** [comment]
@srittau No problem. It was a great learning experience. It covered a lot of concepts.
`def parse_args(self, args: Optional[Sequence[_Text]] = ...) -> Namespace: ...` takes care of case where namespace = None
` def parse_args(self, args: Optional[Sequence[_Text]], namespace: _N) -> _N: ...` this takes case of other values for namespace

1. But I did not understand this overload `def parse_args(self, *, namespace: _N) -> _N: ...` .  Also what does `*` mean ? 
2. When to use Type[_N] and only _N ?

**Pull request merged**